Core/HLE/sceIo.cpp

```diff
@@ -2504,6 +2506,8 @@ static u32 sceIoDread(int id, u32 dirent_addr) {
 		bool isFAT = pspFileSystem.FlagsFromFilename(dir->name) & FileSystemFlags::SIMULATE_FAT32;
 		// Only write d_private for memory stick
 		if (isFAT) {
+			// All files look like they're executable on FAT. This is required for Beats, see issue #14812
+			entry->d_stat.st_mode |= 0111;
```

Comment:
Hm.  I thought we had a test for this against memstick, but maybe it was actually from host0.  However, almost always directories should have the executable bit (some filesystem libs will interpret that correctly as "can traverse" and might restrict to avoid syscall errors.)

Does it fix Beats if it's only directories?

-[Unknown]

Reply:
I see, i guess it's similar to the contents of iOS' IPA(which is a ZIP) where all the folders have executable attribute even when the files in it aren't executable.

Reply:
The Unix/BSD/Linux/etc. rule is that the 1 bits mean execute for files, and mean "search" or "traverse" for directories - in other words, ability to descend into and access files within.  iOS is based on Darwin which is based on BSD, so it inherits the same meaning for those file mode bits.

-[Unknown]

Reply:
I think I tested this, and I confirmed it now - we need the executable flag on the files too, not just the directories.

In fact, we can skip setting execute on the directories and it still works.

Reply:
Huh, strange.

-[Unknown]